torch_xla/csrc/runtime/computation_client.h

```diff
@@ -107,13 +149,41 @@ class ComputationClient {
 
     const xla::ProgramShape& program_shape() const { return program_shape_; }
 
+    const torch::lazy::hash_t& hash() const { return hash_; }
+
+    int parameters_size() const override {
+      return program_shape().parameters_size();
+    }
+
+    const std::vector<torch::lazy::Shape>& parameter_shapes() const override {
+      XLA_ERROR() << "Unimplemented";
+    }
+
+    const std::vector<std::string>& parameter_names() const override {
+      return program_shape().parameter_names();
```

Comment:
seems like we can't alwasy reply on that program_shape exists

Reply:
All of the constructors initialize program shape, as long as its available from the `XlaComputation`. See my other comment.